R/enum.R

```diff
@@ -0,0 +1,105 @@
+enum_value <- function(x, enum_id, all_val) {
+  structure(
+    x,
+    enum_id = enum_id,
+    all_val = all_val,
+    class = "shinyloadtest_enum_value"
+  )
+}
+
+`==.shinyloadtest_enum_value` <- function(x, y) {
+  if (class(y) != "shinyloadtest_enum_value") return(FALSE)
+  identical(x, y) && (attr(x, "enum_id") == attr(y, "enum_id"))
+}
+
+enum_count <- rlang::env(n = 0)
```

Comment:
I believe it's generally not a good idea to have top-level package variables that are objects created by other packages. At shinyloadtest install time, this `rlang::env` object will be serialized, but then rlang could be upgraded at a later date and then you could have a mismatch.

Maybe something like this instead:

```r
enum_counter <- local({
  n <- 0L
  function() {
    (n <<- n + 1L)
  }
})